chainer/reporter.py

```diff
@@ -0,0 +1,325 @@
+        observer object if given.
+
+        Args:
+            values (dict): Dictionary of observed values.
+            observer: Observer object. Its object ID is used to retrieve the
+                observer name, which is used as the prefix of the registration
+                name of the observed value.
+
+        """
+        if observer is not None:
+            observer_id = id(observer)
+            if observer_id not in self._observer_names:
+                raise KeyError(
+                    'Given observer is not registered to the reporter.')
+            observer_name = self._observer_names[observer_id]
+            for key, value in six.iteritems(values):
+                name = '%s/%s' % (observer_name, key)
+                self.observation[name] = value
+        else:
+            self.observation.update(values)
+
+
+_reporters = []
+
+
+def get_current_reporter():
+    """Returns the current reporter object."""
+    return _reporters[-1]
+
+
+def report(values, observer=None):
+    """Reports observed values with the current reporter object.
+
+    Any reporter object can be set current by the ``with`` statement. This
+    function calls the :meth:`Report.report` method of the current report.
+    If no reporter object is current, this function does nothing.
+
+    .. admonition:: Example
+
+       The most typical example is a use within links and chains. Suppose that
+       a link is registered to the current reporter as an observer (for
+       example, the target link of the optimizer is automatically registered to
+       the reporter of the :class:`~chainer.training.Trainer`). We can report
+       some values from the link as follows::
+
+          class MyRegressor(chainer.Chain):
+              def __init__(self, predictor):
+                  super(MyRegressor, self).__init__(predictor=predictor)
+
+              def __call__(self, x, y):
+                  # This chain just computes the mean squared error between the
+                  # prediction and y.
+                  pred = self.predictor(x)
+                  abs_error = F.sum(F.abs(pred - y)) / len(x.data)
+                  loss = F.mean_squared_error(pred, y)
+
+                  # Report the mean absolute and squared error.
+                  report({'abs_error': abs_error, 'squared_error': loss}, self)
+
+                  return loss
+
+       If the link is named ``'main'`` in the hierarchy (which is the default
+       name of the target link in the
+       :class:`~chainer.training.StandardUpdater`), these reported values are
+       named ``'main/abs_error'`` and ``'main/squared_error'``. If these values
+       are reported inside the :class:`~chainer.training.extension.Evaluator`
+       extension, the names are changed to ``'validation/main/abs_error'`` and
+       ``'validation/main/squared_error'`` (``'validation'`` is the default
+       name of the Evaluator extension).
```

Comment:
You mean this link is registered to the reporter as a observer named `validation/main`, don't you?